File: GeneratedEntityReaderClasses/Classes/types_iaabb.cs
namespace GeneratedNoitaClasses;
public class types_iaabb :  iNoitaType<types_iaabb>{
	public NoitaCVector2<NoitaInt> min { get; set; } = new();
	public NoitaCVector2<NoitaInt> max { get; set; } = new();

    public void Read(NoitaStream s){
		min.Read(s);
		max.Read(s);

    }
    public void Write(NoitaStream s){
		min.Write(s);
		max.Write(s);

    }
}

File: GeneratedEntityReaderClassesPython/GeneratedClasses/types_iaabb.py
from dataclasses import dataclass, field
from BasicNoitaTypes import NoitaCVector2
from BasicNoitaTypes import NoitaInt

from NoitaStream import NoitaStream
@dataclass
class types_iaabb:
    min:NoitaCVector2 = field(default_factory=lambda: NoitaCVector2(NoitaInt))
    max:NoitaCVector2 = field(default_factory=lambda: NoitaCVector2(NoitaInt))
    
    def Read(self, s:NoitaStream):
        self.min.Read(s)
        self.max.Read(s)
    
    def Write(self, s:NoitaStream):
        self.min.Write(s)
        self.max.Write(s)

